packages/ui/src/form.tsx

```diff
@@ -140,7 +140,16 @@ const FormMessage = React.forwardRef<
 	React.HTMLAttributes<HTMLParagraphElement>
 >(({ className, children, ...props }, ref) => {
 	const { error, formMessageId, name } = useFormField();
-	const body = error ? String(error?.message) : children;
+	let body = children;
+	if (error) {
+		if (error.message) {
+			body = String(error.message);
+		}
+		if (Array.isArray(error)) {
+			body = error[0]?.message;
```

Comment:
possibly! this would come up for something like a string in a number array field. I believe it would put an error for each value in the array, but there isn't actually a place to put an error message per field value. I think we can leave this as-is for now, and update when we see different kinds of error messages (without this lil modification, it was rendering 'undefined' (the string) as the error message, so if different types of errors come up we should be able to notice it and adjust accordingly)

Reply:
great, thanks for explaining when it happens. i wasn't sure!